src/main/java/nl/tudelft/contextproject/audio/BackgroundMusic.java

```diff
@@ -0,0 +1,212 @@
+package nl.tudelft.contextproject.audio;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.jme3.audio.AudioData.DataType;
+import com.jme3.audio.AudioNode;
+import com.jme3.audio.AudioSource;
+
+import nl.tudelft.contextproject.Main;
+import nl.tudelft.contextproject.logging.Log;
+
+import org.mockito.Mockito;
+
+import lombok.SneakyThrows;
+
+/**
+ * Class for playing/managing background music.
+ */
+public final class BackgroundMusic {
+	private static final BackgroundMusic INSTANCE = new BackgroundMusic();
+	
+	private List<String> music = new ArrayList<String>();
+	private int currentIndex = -1;
+	private AudioNode current;
+	private boolean testing;
+	
+	/**
+	 * Creates a new BackgroundMusic object and loads the music.
+	 */
+	private BackgroundMusic() {
+		loadMusic();
+	}
+	
+	/**
+	 * @return
+	 * 		the BackgroundMusic instance
+	 */
+	public static BackgroundMusic getInstance() {
+		return INSTANCE;
+	}
+	
+	/**
+	 * Retrieves a list of all music files available.
+	 * 
+	 * This method should only be called once.
+	 */
+	@SneakyThrows(URISyntaxException.class)
+	private final void loadMusic() {
+		//Get a list of all music files available.
+		File folder = new File(getClass().getResource("/Sound/Music/").toURI());
```

Comment:
That is not relevant here, getClass().getResource() works with forward slashes, as it usually works in your own jar file. / is the root of the jar, so this code simply gets the Sound/Music folder starting from the root of the jar.